src/devhub/entity/proposal/Proposal.jsx

```diff
@@ -391,21 +395,58 @@ const RadioButton = ({ value, isChecked, label }) => {
 const isAllowedToEditProposal = Near.view(
   "${REPL_DEVHUB_CONTRACT}",
   "is_allowed_to_edit_proposal",
-  { proposal_id: proposal.id, editor: accountId }
+  {
+    proposal_id: proposal.id,
+    editor: accountId,
+  }
+);
+
+const isModerator = Near.view("${REPL_DEVHUB_LEGACY}", "has_moderator", {
+  account_id: accountId,
+});
+
+let grantNotify = Near.view(
+  "${REPL_SOCIAL_CONTRACT}",
+  "is_write_permission_granted",
+  {
+    predecessor_id: "${REPL_DEVHUB_CONTRACT}",
+    key: accountId + "/index/notify",
+  }
 );
 
-const isModerator = isAllowedToEditProposal && proposal.author_id !== accountId;
+const userStorageDeposit = Near.view(
+  "${REPL_SOCIAL_CONTRACT}",
+  "storage_balance_of",
+  {
+    account_id: accountId,
+  }
+);
 
 const editProposalStatus = ({ timeline }) => {
-  Near.call({
-    contractName: "${REPL_DEVHUB_CONTRACT}",
-    methodName: "edit_proposal_timeline",
-    args: {
-      id: proposal.id,
-      timeline: timeline,
+  const calls = [
+    {
+      contractName: "${REPL_DEVHUB_CONTRACT}",
+      methodName: "edit_proposal_timeline",
+      args: {
+        id: proposal.id,
+        timeline: timeline,
+      },
+      gas: 270000000000000,
     },
-    gas: 270000000000000,
-  });
+  ];
+  if (grantNotify === false) {
+    calls.unshift({
+      contractName: "${REPL_SOCIAL_CONTRACT}",
+      methodName: "grant_write_permission",
+      args: {
+        predecessor_id: "${REPL_DEVHUB_CONTRACT}",
+        keys: [accountId + "/index/notify"],
+      },
+      gas: Big(10).pow(14),
+      deposit: getDepositAmountForWriteAccess(userStorageDeposit),
+    });
+  }
```

Comment:
@Megha-Dev-19 I believe that we don't need to grant permissions anymore: https://github.com/near/near-discovery-components/pull/692#pullrequestreview-1926715790

`devhub/mention` and `devhub/edit` are now posted on behalf of devhub.near contract and thus it does not need to have permissions to post on behalf of the user anymore.

`devhub/reply` and `devhub/like` are now originated from the UI and interacts with SocialDB directly, so users will sign their transaction reaching SocialDB directly, so we don't need to setup permissions there either.

Reply:
got it, I will remove it then
so for notifications I will need the editor_id to show the right person profile, rather than devhub.near account, right?
can we pass it in value for notification data?
or should I fetch proposal data using the proposal id?